### PR TITLE
User Stories 17 & 18

### PR DESCRIPTION
**Pull request by dborski**
* Add orderly gem
* Create active_items, popular_items, and quantity_purchased item model methods
* Update items index view and specs to only display active items.
* Add top 5 most popular items to items index view along with amount purchased of each item
* Add top 5 least popular items to index view along with amount purchased of each item
* Create order factory bot

### CODE REVIEW

File: app/controllers/items_controller.rb [hunk 1]
```diff
@@ -3,9 +3,9 @@ class ItemsController<ApplicationController
   def index
     if params[:merchant_id]
       @merchant = Merchant.find(params[:merchant_id])
-      @items = @merchant.items
+      @items = @merchant.items.active_items
```
Comment (perryr16) [inline app/controllers/items_controller.rb:6]: good use of that column

File: app/models/item.rb [hunk 2]
```diff
@@ -25,4 +25,19 @@ def no_orders?
...
+  def self.active_items
+    Item.where(active?: true)
+  end
+
+  def self.popular_items(limit, order)
+    Item.select("items.*, sum(item_orders.quantity)")
+        .joins(:item_orders)
+        .group('items.id')
+        .order("sum(item_orders.quantity) #{order}")
+        .limit(limit)
```
Comment (perryr16) [inline app/models/item.rb:37]: This looks fun

File: app/views/items/index.html.erb [hunk 3]
```diff
@@ -1,25 +1,53 @@
...
-      <% else %>
-        <p>Inactive</p>
-      <% end %>
-    </section>
+  <section class="grid-item" id='item-<%=item.id%>'>
+    <h2> <%=link_to item.name, "/items/#{item.id}" %> </h2>
+    <p>Sold by: <%=link_to item.merchant.name, "/merchants/#{item.merchant.id}" %></p>
+    <%= link_to image_tag("#{item.image}"), "/items/#{item.id}" %>
+    <p> <%= item.description unless @merchant%> </p>
+    <p>Price: <%=number_to_currency(item.price) %> </p>
```
Comment (perryr16) [inline app/views/items/index.html.erb:14]: well named helper method

File: spec/features/items/index_spec.rb [hunk 4]
```diff
@@ -47,15 +47,99 @@
...
-        expect(page).to have_link(@brian.name)
-        expect(page).to have_css("img[src*='#{@dog_bone.image}']")
-      end
+      expect(page).to_not have_link(@dog_bone.name)
+      expect(page).to_not have_content(@dog_bone.description)
+      expect(page).to_not have_content("Price: $#{@dog_bone.price}")
     end
   end
+  describe "When I visit the items index page" do
+    before(:each) do
```
Comment (perryr16) [inline spec/features/items/index_spec.rb:56]: big fan of the `before :each do` statement. it should speed up our specs

File: app/views/items/index.html.erb (hunk 3 above)
Comment (dborski) [reply app/views/items/index.html.erb:14]: I'm not sure, my first guess is that it's a module. Not sure if it's baked in or not.

Comment (dborski) [reply app/views/items/index.html.erb:14]: I looked it up and it is baked in, so that's good to know!

**Comment by foleymichelle9** [comment]
looks good! I approve merging!

**Pull request merged**